src/state/actions/scigateway.actions.tsx

```diff
@@ -490,15 +490,10 @@ export const verifyUsernameAndPassword = (
         dispatch(authorised());
 
         // redirect the user to the original page they were trying to get to
-        // the referrer is added by the redirect in routing.component.tsx
-        const previousRouteState = getState().router.location.state;
```

Comment:
I think if you moved the reading of `previousRouteState` before dispatching the `authorised()` action then the router referrer would be accurate. The redirect to `/logout` happens because if a user tries to access the login page when logged in they should instead see the logout page. If you grab what's in the referrer before the user is logged in, then this should be before the redirect to `/logout`.

Though tbh, I don't mind doing it via localStorage, just thought you'd want to know what was causing the problem :)

Reply:
Oh I think I get it. I'll give it another go